src/Workspaces/CSharp/Portable/Simplification/CSharpSimplificationService.cs

```diff
@@ -22,8 +22,8 @@ internal partial class CSharpSimplificationService : AbstractSimplificationServi
         // 2. Extension method reducer may insert parentheses.  So run it before the parentheses remover.
         private static readonly ImmutableArray<AbstractReducer> s_reducers =
             ImmutableArray.Create<AbstractReducer>(
-                new CSharpCastReducer(),
                 new CSharpNameReducer(),
+                new CSharpCastReducer(),
```

Comment:
@DustinCampbell This was the interesting starting point.  One of our features (convert foreach to for) generates code like so:

```c#
IList<int> list = (IList<int>)collection;
```

Previous behavior was that it wanted this to change into:

```c#
var list = (IList<int>)collection;
```

However, after my change it converted to:

```c#
IList<int> list = collection;
```

This was due to the 'cast' reducer running before the name reducer.  IMO, if someone expresses a 'var' preference, i think that should take preference, so i changed to have name-reduction happen before cast-reduction.   However, this had observable downstream impacts in a few tests that i've called out.